modules/test-ui/src/App.tsx

```diff
@@ -37,40 +37,40 @@ function App() {
   const [transferForm] = Form.useForm();
   const [signMessageForm] = Form.useForm();
 
-  useEffect(() => {
-    const effect = async () => {};
-    effect();
-  }, []);
-
   const connectNode = async (
     iframeSrc: string,
     supportedChains: number[],
     routerPublicIdentifier: string,
   ): Promise<BrowserNode> => {
     try {
       setConnectLoading(true);
+      const chainProviders = {};
+      supportedChains.forEach((chain) => {
+        chainProviders[chain] = config.chainProviders[chain];
+      });
       const client = new BrowserNode({
         supportedChains,
         iframeSrc,
         routerPublicIdentifier,
-        chainProviders: config.chainProviders,
+        chainProviders,
       });
       await client.init();
-      await client.resumePendingCrossChainTransfers();
       const channelsRes = await client.getStateChannels();
       if (channelsRes.isError) {
         setConnectError(channelsRes.getError().message);
         return;
       }
       const channelAddresses = channelsRes.getValue();
-      const _channels = await Promise.all(
-        channelAddresses.map(async (c) => {
-          const channelRes = await client.getStateChannel({ channelAddress: c });
-          console.log("Channel found in store:", channelRes.getValue());
-          const channelVal = channelRes.getValue() as FullChannelState;
-          return channelVal;
-        }),
-      );
+      const _channels = (
+        await Promise.all(
+          channelAddresses.map(async (c) => {
+            const channelRes = await client.getStateChannel({ channelAddress: c });
+            console.log("Channel found in store:", channelRes.getValue());
+            const channelVal = channelRes.getValue() as FullChannelState;
+            return channelVal;
+          }),
+        )
+      ).filter((chan) => supportedChains.includes(chan.networkContext.chainId));
```

Comment:
Not sure if we should do this in test-ui, what if we don't have supported chains (everything should work without it).

Reply:
This was causing errors for me when i only wanted to connect to one provider.

Transfers won't work without it, since you can't call chain service functions (i.e. `create` or `resolve`)